### PR TITLE
Output warning info when series is invalid

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

To help diagnose issues when the distro-info is out of date, we
print warnings to the command line to help operators understand
what the potential cause of the issue is.

In the PR we print out what warnings where made when parsing the
os series package. This should help operators diagnose the fact their
system is out of date and needs updating.

  - [x] Update dependency once `juju/os` PR has been approved.

## QA steps

The easiest way is to edit `/usr/share/distro-info/ubuntu.csv` and change
the Trusty line to the following:

```14.04 LTS,Trusty Tahr,trusty,2013-10-17,2014-04-17```

Ensure you have a bootstrapped controller and run the following:

`juju deploy cs:trusty/glance`

It should output the following:

```console
WARNING Parsing issues occured when extracting the distro-info.
You may want to try updating your distro-info using `apt-get update distro-info`
  - EOL date not found, falling back to release date, plus 5 years
  - EOL ESM date not found, falling back to EOL date
ERROR glance is not available on the following series: trusty not supported
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1833763

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/urMhXc2R/1565-1833763-investigate-trusty-not-deploying

**Comment by hmlanigan** [comment]
!!build!!

File: cmd/juju/application/deploy.go [hunk 1]
```diff
@@ -1650,12 +1653,38 @@ func (c *DeployCommand) charmStoreCharm() (deployFn, error) {
...
+
+// charmValidationError consumes an error along with a charmSeries and name
+// to help provide better feedback to the user when somethings gone wrong around
+// validating a charm validation
+func charmValidationError(charmSeries, name string, err error) error {
+	if err != nil {
 		if errors.IsNotSupported(err) {
+			if warningInfo := series.SeriesWarningInfo(charmSeries); len(warningInfo) > 0 {
+				logger.Warningf(
+					"Parsing issues occured when extracting the distro-info.\n"+
```
Comment (hmlanigan) [inline cmd/juju/application/deploy.go]: s/occured/occurred/  or is that a british-ism?

File: cmd/juju/application/deploy.go [hunk 2]
```diff
@@ -1650,12 +1653,38 @@ func (c *DeployCommand) charmStoreCharm() (deployFn, error) {
...
+	return charmValidationError(series, name, errors.Trace(err))
+}
+
+// charmValidationError consumes an error along with a charmSeries and name
+// to help provide better feedback to the user when somethings gone wrong around
+// validating a charm validation
+func charmValidationError(charmSeries, name string, err error) error {
+	if err != nil {
 		if errors.IsNotSupported(err) {
+			if warningInfo := series.SeriesWarningInfo(charmSeries); len(warningInfo) > 0 {
```
Comment (hmlanigan) [inline cmd/juju/application/deploy.go]: suggest adding the bug number in comments, this is a very specific case.

File: cmd/juju/application/deploy.go (hunk 1 above)
Comment (SimonRichardson) [reply cmd/juju/application/deploy.go]: dunno, internet isn't telling me :D

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**